src/components/ARDownloadListItem.js

```diff
@@ -0,0 +1,100 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { ActivityIndicator, Alert, StyleSheet } from 'react-native';
+import { ListItem } from 'react-native-elements';
+
+import { colors, consts, Icon, normalize, texts } from '../config';
+import { deleteObject, downloadObject, DOWNLOAD_TYPE, formatSize } from '../helpers';
+
+import { RegularText } from './Text';
+import { Touchable } from './Touchable';
+
+const BUTTON_SIZE = normalize(16);
+
+export const ARDownloadListItem = ({ item, index, data, setData }) => {
+  const { DOWNLOAD_TYPE: itemDownloadType, progressSize, size, title, totalSize } = item;
+
+  const onPress = async () => {
+    if (itemDownloadType === DOWNLOAD_TYPE.DOWNLOADABLE) {
+      const { newDownloadedData } = await downloadObject({
+        item,
+        index,
+        downloadableData: data,
+        setDownloadableData: setData
+      });
+      setData(newDownloadedData);
+    } else if (itemDownloadType === DOWNLOAD_TYPE.DOWNLOADED) {
+      Alert.alert(
+        texts.settingsTitles.arListLayouts.alertTitle,
+        texts.settingsTitles.arListLayouts.deleteAlertMessage,
+        [
+          {
+            text: texts.settingsTitles.arListLayouts.cancel,
+            style: 'cancel'
+          },
+          {
+            text: texts.settingsTitles.arListLayouts.deleteAlertButtonText,
+            onPress: async () => {
+              const { newDownloadedData } = await deleteObject({
+                item,
+                index,
+                downloadableData: data
+              });
+              setData(newDownloadedData);
+            },
+            style: 'destructive'
+          }
+        ]
+      );
+    } else {
+      return;
+    }
+  };
+
+  return (
+    <ListItem
+      title={<RegularText small>{title}</RegularText>}
+      subtitle={() => subtitleRender({ totalSize, size, progressSize })}
+      bottomDivider
+      containerStyle={styles.container}
+      rightIcon={() =>
+        itemDownloadType === DOWNLOAD_TYPE.DOWNLOADED ? (
+          <Icon.CloseCircle color={colors.darkText} size={BUTTON_SIZE} />
+        ) : itemDownloadType === DOWNLOAD_TYPE.DOWNLOADABLE ? (
+          <Icon.ArrowDownCircle color={colors.primary} size={BUTTON_SIZE} />
+        ) : (
+          <ActivityIndicator size="small" />
+        )
+      }
+      onPress={onPress}
+      delayPressIn={0}
+      Component={Touchable}
+      accessibilityLabel={`(${title}) ${consts.a11yLabel.button}`}
+    />
+  );
+};
+
+const subtitleRender = ({ totalSize, size, progressSize }) => {
+  return progressSize ? (
+    <RegularText smallest>{`${
+      size >= totalSize ? formatSize(size) : formatSize(progressSize)
+    } / ${formatSize(totalSize)}`}</RegularText>
```

Comment:
This is because we do not download a single file. We are trying to download multiple files in one go. There needs to be different progress for each data. For this reason, it appears more than once on the screen.

Reply:
hm ok. let's think of a solution for this to nit have that "jumping"

Reply:
I'll do some research on it. 👍

Reply:
I've searched, but I haven't found anything. 😔

Reply:
in the new PR i still saw some jumpings at the end of a download.

Reply:
we were able to fix it in a live share with correct calculation of the `size`  inside of the data json stored on the device.